webui/src/app/core/providers.resource.js

```diff
@@ -4,44 +4,41 @@ var angular = require('angular');
 var traefikCoreProvider = 'traefik.core.provider';
 module.exports = traefikCoreProvider;
 
-angular
-  .module(traefikCoreProvider, ['ngResource'])
-  .factory('Providers', Providers);
+angular.module(traefikCoreProvider, ['ngResource']).factory('Providers', Providers);
 
 /** @ngInject */
-function Providers($resource) {
+function Providers($resource, $q) {
   const resourceProvider = $resource('../api/providers');
   return {
-    get: function () {
-      const rawProviders = resourceProvider.get();
-
-      for (let providerName in rawProviders) {
-        if (rawProviders.hasOwnProperty(providerName)) {
-
-          // BackEnds mapping
-          let bckends = rawProviders[providerName].backends;
-
-          rawProviders[providerName].backends = Object.keys(bckends)
-            .map(key => {
-              const goodBackend = bckends[key];
-              goodBackend.backendId = key;
-              return goodBackend;
-            });
-
-          // FrontEnds mapping
-          let frtends = rawProviders[providerName].frontends;
-
-          rawProviders[providerName].frontends = Object.keys(frtends)
-            .map(key => {
-              const goodFrontend = frtends[key];
-              goodFrontend.frontendId = key;
-              return goodFrontend;
-            });
-
-        }
-      }
-
-      return rawProviders;
+    get: function() {
+      return $q((resolve, reject) => {
```

Comment:
$q is the preferred way to handle promises with AngularJS (1.X) , since it wraps the promise mechanism into a $digest loop, which allows immediate view refresh. For instance, notice how it takes 2 additional seconds to display data on startup, before it effectively does the $interval call to that refreshes data.